client/src/entrypoints/admin/modal-workflow.js

```diff
@@ -109,10 +126,14 @@ function ModalWorkflow(opts) {
   };
 
   self.close = function () {
-    container.modal('hide');
+    if (useDialog) {
+      self.dialog.dispatchEvent(new CustomEvent('hide'));
```

Comment:
This isn’t documented formally anywhere but we’ve discussed having a `wagtail:` for all custom events

```suggestion
      self.dialog.dispatchEvent(new CustomEvent('wagtail:hide'));
```